user/tlu/module/src/TluRawEvent2StdEventConverter.cc

```diff
@@ -29,41 +29,142 @@ bool TluRawEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::Standa
     d2->SetTag(TLU+stm+"_TRG", std::to_string(d1->GetTriggerN()));
   }
 
-  auto triggersFired = std::stoi(d1->GetTag("TRIGGER" , "0"), nullptr, 2); // interpret as binary
-  auto finets0 = std::stoi(d1->GetTag("FINE_TS0", "0"));
-  auto finets1 = std::stoi(d1->GetTag("FINE_TS1", "0"));
-  auto finets2 = std::stoi(d1->GetTag("FINE_TS2", "0"));
-  auto finets3 = std::stoi(d1->GetTag("FINE_TS3", "0"));
-  auto finets4 = std::stoi(d1->GetTag("FINE_TS4", "0"));
-  auto finets5 = std::stoi(d1->GetTag("FINE_TS5", "0"));
+  uint8_t triggersFired;
+  uint32_t finets0;
+  uint32_t finets1;
+  uint32_t finets2;
+  uint32_t finets3;
+  uint32_t finets4;
+  uint32_t finets5;
 
-  // Set times for StdEvent in picoseconds (timestamps provided in nanoseconds):
-  // Note: __builtin_popcount counts the "ones" in a binary, see here:
-  // https://www.geeksforgeeks.org/count-set-bits-in-an-integer/
-  // I.e. add up all timestamps for which the triggersFired bit is 1 and divide by number of active triggers:
-  // Factor 25./32. -> fine timestamp in 781ps binning
-  auto finets = ((finets0 * ((triggersFired >> 0) & 0x1))
-               + (finets1 * ((triggersFired >> 1) & 0x1))
-               + (finets2 * ((triggersFired >> 2) & 0x1))
-               + (finets3 * ((triggersFired >> 3) & 0x1))
-               + (finets4 * ((triggersFired >> 4) & 0x1))
-               + (finets5 * ((triggersFired >> 5) & 0x1)))
-               / __builtin_popcount(triggersFired & 0x3F); // count "ones" in binary
-
-  auto ts = (d1->GetTimestampBegin() * 1000) & 0xFFFFFFFFFFFFFF00 + static_cast<uint64_t>(25. / 32. * finets);
+  // Should I read these in at m_first_time and store in static member variable or read in each time?
```

Comment:
Have you checked what the penalty in computing time is? In any case, please remove the comment :)

Reply:
@jekroege fix comment

Haven't checked penalty but it's offline, so it shouldn't really matter. The whole converter concept with static instances is a bit flawed anyway... ;)

Reply:
Yep, just saw it, too...will  be fixed in the next MR I'm pushing in 5min :)